core/state_processor.go

```diff
@@ -199,3 +203,29 @@ func ApplyTransaction(config *params.ChainConfig, bc ChainContext, author *commo
 	vmenv := vm.NewEVM(blockContext, vm.TxContext{}, statedb, config, cfg)
 	return applyTransaction(msg, config, bc, author, gp, statedb, header.Number, header.Hash(), tx, usedGas, vmenv)
 }
+
+// ProcessParentBlockHash stores the parent block hash in the history storage contract
+// as per EIP-2935.
+func ProcessParentBlockHash(prevHash common.Hash, vmenv *vm.EVM, statedb *state.StateDB) {
+	// TODO: consider tracers
```

Comment:
Is EIP-2935 tracing-related `TODO`s (also in API functions) optional?

Reply:
Nevermind, just saw that there are some block-level tracers in upstream, added in [6a01e5a](https://github.com/scroll-tech/go-ethereum/pull/1193/commits/6a01e5af3b13aa0ddf554295f00e4c8f3269c5c2).